File: backend/src/main/java/com/viniciusfinger/rsflood/dto/NecessityDTO.java
package com.viniciusfinger.rsflood.dto;

import com.viniciusfinger.rsflood.enums.NecessityCategory;

import lombok.Data;

@Data
public class NecessityDTO {

    private Long id;

    private String description;

    private Long quantity;

    private NecessityCategory category;
}

File: frontend/src/app/model/dto/necessity-dto.ts
import { NecessityCategory } from "../enum/necessity-category";

export class NecessityDTO {
    id: number;
    description: string;
    quantity: number;
    category: NecessityCategory;

    constructor(id: number, description: string, quantity: number, category: NecessityCategory) {
        this.id = id;
        this.description = description;
        this.quantity = quantity;
        this.category = category;
    }
}

